lib/ohai/plugins/joyent.rb

```diff
@@ -25,21 +25,18 @@
   depends 'os', 'platform', 'virtualization'
 
   def collect_product_file
-    lines = []
-    if ::File.exists?("/etc/product")
-      ::File.open("/etc/product") do |file|
-        while line = file.gets
-          lines << line
-        end
-      end
+    data = ::File.read("/etc/product") rescue nil
```

Comment:
You'll need to stub `::File.read("/etc/product")` in the specs. The failures appear to be due to [this line](https://github.com/bahamas10/ohai/blob/dave-1426559693/lib/ohai/plugins/joyent.rb#L62) returning `nil`.